lib/rust/ensogl/app/theme/derive/src/from_theme.rs

```diff
@@ -120,13 +118,9 @@ fn make_struct_inits(
             .ident
             .as_ref()
             .expect("Encountered unnamed struct field. This cannot not happen.");
-        let init = match ThemeTypes::from_ty(&field.ty) {
-            ThemeTypes::Color | ThemeTypes::Number => quote! {
-                #field_name:*#field_name,
-            },
-            ThemeTypes::String => quote! {
-                #field_name:#field_name.clone(),
-            },
+        // Keep in mind that all [`Copy`] types also implement [`Clone`].
+        let init = quote! {
+            #field_name : #field_name.clone(),
```

Comment:
FromTheme macro now supports arbitrary types, given a corresponding `accessor` is provided. I believe this simplification is valid.